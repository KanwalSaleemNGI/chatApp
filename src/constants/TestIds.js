export default {
  authButton: {
    title: 'title',
    onPress: 'onPress',
  },
  chatInfo: {
    onPress: 'onPress',
  },
  login: {
    emailInput: 'emailInput',
    passwordInput: 'passwordInput',
  },
};
