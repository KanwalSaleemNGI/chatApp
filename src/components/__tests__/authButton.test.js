import React from 'react';
import {AuthButton} from '../';
import {fireEvent, render} from '@testing-library/react-native';
import TestIds from '../../constants/TestIds';

describe('AuthButton comp', () => {
  const authButtonTitle = 'Login';

  const onPress = jest.fn();
  console.log(onPress);
  const renderHandler = () => {
    return render(<AuthButton onPress={onPress}>{authButtonTitle}</AuthButton>);
  };

  it('should match snapshot', () => {
    //arrange
    const {toJSON} = renderHandler();

    //act

    //assert
    expect(toJSON()).toMatchSnapshot();
  });

  it('should properly render the AuthButton title', () => {
    //arrange
    const {getByTestId} = renderHandler();

    //act

    //assert
    expect(getByTestId(TestIds.authButton.title).props.children).toEqual(
      authButtonTitle,
    );
  });

  it('should press AuthButton', () => {
    //arrange
    const {getByTestId} = renderHandler();

    //act
    fireEvent(getByTestId(TestIds.authButton.onPress), 'press');

    //assert
    expect(onPress).toHaveBeenCalled();
  });
});
