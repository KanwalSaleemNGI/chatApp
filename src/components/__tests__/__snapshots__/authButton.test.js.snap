// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthButton comp should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4C96D6",
      "borderRadius": 25,
      "elevation": 5,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 8,
      "width": "90%",
    }
  }
  testID="onPress"
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "color": "#fff",
        "fontFamily": "Roboto-Regular",
        "fontSize": 16,
      }
    }
    testID="title"
  >
    Login
  </Text>
</View>
`;
